pycytominer/cyto_utils/DeepProfiler_processing.py

```diff
@@ -0,0 +1,160 @@
+            file_key = file_info[self.aggregate_on]
+
+            if self.aggregate_on == "site":
+                file_key = (
+                    f"{file_info['plate']}_{file_info['well']}_{file_info['site']}"
+                )
+
+            if self.aggregate_on == "well":
+                file_key = f"{file_info['plate']}_{file_info['well']}"
+
+            if file_key in self.file_aggregate:
+                self.file_aggregate[file_key]["files"].append(filename)
+            else:
+                self.file_aggregate[file_key] = {}
+                self.file_aggregate[file_key]["files"] = [filename]
+
+            self.file_aggregate[file_key]["metadata"] = file_info
+
+    def aggregate_deep(self):
+        if not hasattr(self, "file_aggregate"):
+            self.setup_aggregate()
+
+        self.aggregated_profiles = []
+        self.aggregate_merge_col = f"Metadata_{self.aggregate_on.capitalize()}_Position"
+
+        for metadata_level in self.file_aggregate:
+            df = pd.concat(
+                [load_npz(x) for x in self.file_aggregate[metadata_level]["files"]]
+            )
+            meta_df = pd.DataFrame(
+                self.file_aggregate[metadata_level]["metadata"], index=[0]
+            )
+            meta_df.columns = [f"Metadata_{x.capitalize()}" for x in meta_df.columns]
+
+            if self.aggregate_on == "well":
+                meta_df = meta_df.drop("Metadata_Site", axis="columns")
+
+            features = df.columns.tolist()
+            df = df.assign(Metadata_Aggregate_On=self.aggregate_on)
+            df = aggregate(
+                population_df=df,
+                strata="Metadata_Aggregate_On",
+                features=features,
+                operation=self.aggregate_operation,
+            )
+            df.loc[:, self.aggregate_merge_col] = metadata_level
+            df = meta_df.merge(df, left_index=True, right_index=True)
+            self.aggregated_profiles.append(df)
+
+        self.aggregated_profiles = pd.concat([x for x in self.aggregated_profiles])
+        self.aggregated_profiles.columns = [
+            str(x) for x in self.aggregated_profiles.columns
+        ]
+        meta_features = infer_cp_features(self.aggregated_profiles, metadata=True)
+        reindex_features = [str(x) for x in features]
+        self.aggregated_profiles = self.aggregated_profiles.reindex(
+            meta_features + reindex_features, axis="columns"
+        )
+
+    def annotate_deep(
+        self, annotate_cols, merge_cols=["Metadata_Plate", "Metadata_Well"]
+    ):
+        if not hasattr(self, "aggregated_profiles"):
+            self.aggregate_deep()
+
+        meta_df = self.index_df.loc[:, annotate_cols].drop_duplicates()
+
+        meta_df.columns = [
+            "Metadata_{}".format(x) if not x.startswith("Metadata_") else x
+            for x in meta_df.columns
+        ]
+
+        return meta_df.merge(self.aggregated_profiles, on=merge_cols, how="inner")
```

Comment:
I see that there is some metadata annotation happening in `load_npz()`. I don't think I have grasped the difference between what is happening there and what is happening in `annotate_deep()`. Can you explain what is different?

Reply:
See discussion below on inconsistencies